testsuite/python/lb_circular_couette.py

```diff
@@ -0,0 +1,186 @@
+        for normal, dist in zip(normals, dists):
+            self.system.lbboundaries.add(espressomd.lbboundaries.LBBoundary(
+                shape=espressomd.shapes.Wall(normal=normal, dist=dist),
+                velocity=0.3 * np.cross(self.params['axis'], normal)))
+        # inner cylinder without slip velocity
+        self.system.lbboundaries.add(espressomd.lbboundaries.LBBoundary(
+            shape=espressomd.shapes.Cylinder(
+                center=self.system.box_l / 2.0, axis=self.params['axis'],
+                direction=1, radius=1., length=BOX_L * 1.5)))
+
+        # the system needs to be fully symmetric
+        mask = np.copy(lbf[:, :, :].boundary.astype(bool))
+        np.testing.assert_array_equal(mask, np.flip(mask, axis=0))
+        np.testing.assert_array_equal(mask, np.flip(mask, axis=1))
+        np.testing.assert_array_equal(mask, np.flip(mask, axis=2))
+
+        # the system needs to be closed in the x and y directions
+        np.testing.assert_array_equal(mask[0, :, :], 1)
+        np.testing.assert_array_equal(mask[-1, :, :], 1)
+        np.testing.assert_array_equal(mask[:, 0, :], 1)
+        np.testing.assert_array_equal(mask[:, -1, :], 1)
+
+        ctp = espressomd.math.CylindricalTransformationParameters(
+            center=[BOX_L / 2.0, BOX_L / 2.0, 0.0],
+            axis=self.params['axis'],
+            orientation=self.params['orientation'])
+        local_obs_params = OBS_PARAMS.copy()
+        local_obs_params['transform_params'] = ctp
+        obs = espressomd.observables.CylindricalLBVelocityProfile(
+            **local_obs_params)
+
+        # simulate until profile converges
+        mid_indices = [int((EFFECTIVE_RADIUS / AGRID) / 2) - 2,
+                       int((BOX_L / AGRID) / 2), int((BOX_L / 2. / AGRID) / 2)]
+        diff = float("inf")
+        old_val = lbf[mid_indices].velocity[1]
+        while diff > 1e-6:
+            self.system.integrator.run(10)
+            new_val = lbf[mid_indices].velocity[1]
+            diff = abs(new_val - old_val)
+            old_val = new_val
+
+        r = obs.bin_centers()[:, :, :, 0].reshape(-1)
+        v_r, v_phi, v_z = np.copy(obs.calculate()).reshape([-1, 3]).T
+
+        def taylor_couette_kernel(r, a, b):
+            return a * r + b / r
+
+        # check velocity is zero for the radial and axial components
+        np.testing.assert_allclose(v_r, 0., atol=1e-3)
+        np.testing.assert_allclose(v_z, 0., atol=1e-8)
+
+        # get Taylor-Couette coefficients
+        (a_sim, b_sim), _ = scipy.optimize.curve_fit(
+            taylor_couette_kernel, r, v_phi)
+        # check deviations are small (the outer cylinder isn't perfect)
+        v_phi_fit = taylor_couette_kernel(r, a_sim, b_sim)
+        np.testing.assert_allclose(v_phi, v_phi_fit, atol=1e-2)
+        # check the correct coefficients are recovered
+        a_ref, b_ref = (-0.0388038, 0.019088255)
+        np.testing.assert_allclose([a_sim, b_sim], [a_ref, b_ref], atol=1e-6)
```

Comment:
@christophlohrmann This test is a backport of the `walberla` branch circular Couette test. Since we don't have per-node slip velocity conditions on the `python` branch, I approximated the outer cylinder with 8 moving walls. I hardcoded the reference coefficients here because the Taylor-Couette equations seems unstable when the outer cylinder rotates instead of the inner cylinder. Let me know if you think there is a better way to write this test.

Reply:
The reference values are obtained using the walberla branch with the per-node boundary condition?

Reply:
Tbh I am surprised the coefficients match to 1e-6 with the octagon construction. Seems suspiciously good.
Also why do we not check against the analytic coefficients? a_ref and b_ref should be calculated

Reply:
This is now tested against the analytical formula. The deviation is 1e-3 due to the inaccuracy of the octagon.